### PR TITLE
[model] add example model: wan

### PR DESCRIPTION
**Pull request by forrestl111**
## 📌 Description

This PR implement one of the most open sourced video generation, Wan, with efficient FlashInfer API.

The motivation is to provide a best practice to show how to improve Wan's performance by simply apply FlashInfer's API. This is useful for users who don't use SGLang/vLLM for video generation deployment, they need a good example to know how to call FlashInfer's API for this model. Also, it can be used to compare the performance between variant FlashInfer's APIs, e.g., NVFP4 gemm/FP8 gemm/etc.

## 🔍 Related Issues

None

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

## Reviewer Notes

I would like to add models in the futures. Would be great if you have any advice about how to organize these models in the future.

## Summary by CodeRabbit

* **New Features**
  * Added a FlashInfer-optimized Wan transformer with configurable GEMM and attention backends, online/offline activation quantization, and optional skip-softmax sparse attention
  * Added a pipeline script to run the model with runtime backend/quantization and export options

* **Documentation**
  * Added a detailed guide describing backend selection, environment overrides, sparse-attention controls, and example usage walkthrough

### CODE REVIEW

File: examples/pytorch/wan/transformer_wan_flashinfer.py [hunk 1]
```diff
@@ -0,0 +1,2715 @@
...
+
+        # gemm_fp8_nt_groupwise: a(m,k) @ b(n,k)^T -> (m, n)
+        # a_scale: (m, k//128) contiguous — for MN-major this is (k_blocks, m).T
+        # b_scale: MN-major -> (k_blocks, n_blocks)
+        out = gemm_fp8_nt_groupwise(
+            x_fp8,
+            self._weight_fp8_nt,
+            x_scale,
+            self._weight_scale_nt,
+            scale_major_mode=None,
```
Comment (saltyminty) [inline examples/pytorch/wan/transformer_wan_flashinfer.py]: I believe gemm_fp8_nt_groupwise with cutlass backend requires a non-None scale_major_mode

File: examples/pytorch/wan/transformer_wan_flashinfer.py [hunk 2]
```diff
@@ -0,0 +1,2715 @@
...
+                return "trtllm"
+            else:
+                import warnings
+
+                fallback = (
+                    self.single_attention_backend
+                    if batch_size == 1
+                    else self.batch_attention_backend
+                )
+                warnings.warn(
```
Comment (saltyminty) [inline examples/pytorch/wan/transformer_wan_flashinfer.py]: In some tests, batch_attention_backend is configured as 'trtllm' – could this lead to a failure due to trtllm trying to fallback onto itself?

File: examples/pytorch/wan/transformer_wan_flashinfer.py [hunk 3]
```diff
@@ -0,0 +1,1446 @@
...
+        field: _config_value(config, field, getattr(WanTransformer3DConfig, field))
+        for field in WanTransformer3DConfig.__dataclass_fields__
+        if field != "inner_dim"
+    }
+
+    attention_backend = os.getenv("FLASHINFER_ATTENTION_BACKEND")
+    if attention_backend:
+        values["attention_backend"] = attention_backend
+
+    for field, env_name in _FLASHINFER_ENV_OVERRIDES.items():
```
Comment (saltyminty) [inline examples/pytorch/wan/transformer_wan_flashinfer.py:198]: NIT: is there any particular reason `FLASHINFER_ATTENTION_BACKEND` has it's own separate logic rather than being as part of `_FLASHINFER_ENV_OVERRIDES`?

File: examples/pytorch/README.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+- Unsupported GEMM backends fall back to `torch` with a warning that lists the
+  required SM range.
+- Skip-softmax sparse attention requires Blackwell (SM100/SM103) and the
+  TRT-LLM attention path.
+- The `trtllm` attention backend is Blackwell-only. On other architectures the
+  dispatcher warns and falls back to `single`/`cudnn` automatically.
+- `from_pretrained` remaps diffusers' `ffn.net.0.proj.*` / `ffn.net.2.*` keys
+  to `FlashInferFeedForward`'s `proj_up.*` / `proj_down.*`; without the remap
+  the FFN weights would be silently dropped.
+- `pipeline_wan_flashinfer.py --compare-original` compares latent tensors by
```
Comment (saltyminty) [inline examples/pytorch/README.md]: Is this argument removed/outdated?

**Review by saltyminty — APPROVED** [review]

File: examples/pytorch/wan/transformer_wan_flashinfer.py (hunk 3 above)
Comment (forrestl111) [reply examples/pytorch/wan/transformer_wan_flashinfer.py:198]: You are right! `FLASHINFER_ATTENTION_BACKEND` should be a part of `_FLASHINFER_ENV_OVERRIDES`

File: examples/pytorch/README.md (hunk 4 above)
Comment (forrestl111) [reply examples/pytorch/README.md]: Yes, this argument is removed.

**Review by saltyminty — APPROVED** [review]
